connectivity.py

```diff
@@ -0,0 +1,100 @@
+from threading import RLock, Event
+
+from nio.properties import TimeDeltaProperty
+from nio.modules.scheduler import Job
+from nio.signal.status import BlockStatusSignal
+from nio.util.runner import RunnerStatus
+
+
+class PubSubConnectivity(object):
+    """ Adds connectivity awareness to pubsub blocks
+    """
+    timeout = TimeDeltaProperty(title='Connect Timeout',
+                                default={'seconds': 2},
+                                advanced=True)
+
+    def __init__(self):
+        super().__init__()
+        self._connected = None
+        self._connected_lock = RLock()
+        self._timeout_job = None
+        self._status_set = {RunnerStatus.warning: False,
+                            RunnerStatus.error: False}
+        self._connected_event = Event()
+
+    def conn_configure(self, is_connected):
+        """ Sets up instance for connectivity checks
+
+        Args:
+            is_connected (callable): function to invoke to establish initial
+                connectivity status
+        """
+        with self._connected_lock:
+            self._connected = is_connected()
+            self.logger.info("Starting in: '{}' state".format(
+                "connected" if self._connected else "disconnected"))
+
+        if not self._connected:
```

Comment:
With the comment below for achieving proper thread safety, since this variable is accessed via multiple threads, we should pull the value within the `self._connected_lock` as well.